torch/distributed/_checkpoint/metadata.py

```diff
@@ -0,0 +1,45 @@
+from typing import Any, Dict, List, Union, Optional
+import torch
+from dataclasses import dataclass
+import pickle
+from torch.distributed._sharded_tensor import ShardedTensor, ShardedTensorMetadata, ShardMetadata
+
+TENSOR_TYPE = Union[torch.Tensor, ShardedTensor]
+
+@dataclass
+class StorageMetadata:
+    shard_metadata: Optional[ShardMetadata]
```

Comment:
ShardMetadata is repeated in both StorageMetadata and TensorMetadata. Could it be isolated to the TensorMetadata only (since they are both contained in ExtendedTensorMetadata. It seems the ExtendedTensorMetadata is the overarching metadata container for saving/loading tensors whether they are distributed or not

Reply:
One is ShardMetadata the other is ShardedTensorMetadata, they are both define in "shard.sharded_tenosr". IMO the name can be confusing, might worth some update, @pritamdamania, @wanchaol